sdk/digitaltwins/Azure.DigitalTwins.Core/src/Models/DigitalTwinMetadataJsonConverter.cs

```diff
@@ -50,6 +49,11 @@ public override DigitalTwinMetadata Read(ref Utf8JsonReader reader, Type typeToC
                 {
                     metadata.PropertyMetadata[propertyName] = JsonSerializer.Deserialize<DigitalTwinPropertyMetadata>(ref reader, options);
                 }
+                else
+                {
+                    // Unexpected
+                    throw new JsonException();
+                }
```

Comment:
Just to confirm, is this the behavior we want?

Imagine the JSON input is:
`"{\"foo\":[], \"$model\": \"some_id\"}`

Do we want the converter to throw, because the value of the property `foo` is not a JSON object?

Similarly for, say, the following where the token of a property that doesn't match `$model` is a string:
`"{\"$model\": \"some_id\", \"bar\": \"some user string\"}`

We want to throw here, correct? Do these JSON input strings capture some valid user scenario?

Reply:
The only valid JSON for this scenario is one where it has a `$model` property, and additional properties where the name matches a custom name that matches one of the DTDL properties (we don't verify that) and the value is an object that has a `lastUpdateTime` property.

It can never be an array, or a property with some other simple data type, to address your specific examples.